lib/internal/modules/esm/resolve.js

```diff
@@ -765,6 +769,27 @@ function resolveAsCommonJS(specifier, parentURL) {
 
 function defaultResolve(specifier, context = {}, defaultResolveUnused) {
   let { parentURL, conditions } = context;
+  if (manifest) {
+    const redirects = manifest.getRedirector(parentURL);
+    if (redirects) {
+      const { resolve, reaction } = redirects;
+      const destination = resolve(specifier, new SafeSet(conditions));
```

Comment:
Is the copy necessarily because of potential mutations? If so, would it make sense to freeze the conditions set? Or is that a perf concern?

Reply:
just to keep the signature of resolve consistent